### PR TITLE
Facility deletion management command

### PR DESCRIPTION
**Pull request by bjester**
### Summary

- Adds a management command for facility deletion to remove a facility entirely from the device
- Adds a `facilitytask` endpoint for initiating a facility deletion task
- Explicitly added the `on_delete=CASCADE` to the models since it in future versions of Django, it requires an explicit action it instead of assuming cascade
- Updated `@action` decorators with preferred alternative since it is deprecated

### Reviewer guidance

A nice to have feature was to have the command report progress. Relying on CASCADE deletions works just fine for most of the models, but also doesn't give much insight into that progress, as far as I know. This currently goes through all the models, counts them, and then deletes them; through this procedure it can report progress since it's handling the models individually.

Some feedback on whether we care about any of that would be great. It was a nice to have that didn't seem too difficult, but could get stale in the future if adding more related models, although still completely deleting the facility because of the cascade.

### References

[Notion page](https://www.notion.so/learningequality/Backend-Admins-can-delete-a-Facility-and-all-its-data-from-a-Device-13-pts-0-14-dev-fac-UC6-2277c6a5cf3940ecaf822f3e0d25bb38)

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] ~If this includes an internal dependency change, a link to the diff is provided~

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] ~If there are any front-end changes, before/after screenshots are included~
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by jonboiser — APPROVED** [review]

File: kolibri/core/auth/management/utils.py [hunk 1]
```diff
@@ -22,6 +23,21 @@
...
+        post_delete.receivers = []
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        post_delete.receivers = self.receivers
+        self.receivers = None
+
+
 def _interactive_client_facility_selection():
     facilities = Facility.objects.all().order_by("name")
     message = "Please choose a facility to sync:\n"
```
Comment (radinamatic) [inline kolibri/core/auth/management/utils.py:43]: @bjester Is the use of `sync` here intentional? This command should delete the facility, so why _sync_...? :thinking:

Comment (bjester) [reply kolibri/core/auth/management/utils.py:43]: @radinamatic Oh good catch! I didn't notice it said "sync". I'll make that dynamic

Comment (bjester) [reply kolibri/core/auth/management/utils.py:43]: Updated here https://github.com/learningequality/kolibri/pull/7187/commits/a41de91333175f594d4bf2d7ce1adaf0a7ed7bd5

File: kolibri/core/auth/management/commands/deletefacility.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+    :type things: list
+    :type size: int
+    """
+    for i in range(0, len(things), size):
+        yield things[i : i + size]
+
+
+class Command(AsyncCommand):
+    help = "This command initiates the deletion process for a facility and all of it's related data."
```
Comment (jamalex) [inline kolibri/core/auth/management/commands/deletefacility.py:138]: *its

File: kolibri/core/auth/management/commands/deletefacility.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+        if not noninteractive:
+            # ensure the user REALLY wants to do this!
+            confirm_or_exit(
+                "Are you sure you wish to permanently delete this facility? This will DELETE ALL DATA FOR THE FACILITY."
+            )
+            confirm_or_exit(
+                "ARE YOU SURE? If you do this, there is no way to recover the facility data on this device."
+            )
+
+        # everything should get cascade deleted from the facility, but we'll check anyway
```
Comment (jamalex) [inline kolibri/core/auth/management/commands/deletefacility.py:175]: Sounds good. The main thing that I could imagine might be less likely to cascade delete would be anonymous ContentSessionLogs, which don't FK onto a user. But they would still have an FK onto the FacilityDataset (as would all of these), so it _should_ all get cascade deleted, yeah.

File: kolibri/core/auth/management/commands/deletefacility.py [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+            total_deleted = 0
+
+            # run the counting step
+            with self.start_progress(
+                total=delete_group.group_count()
+            ) as update_progress:
+                update_progress(increment=0, message="Counting database objects")
+                total_count = delete_group.count(update_progress)
+
+            # no the deleting step
```
Comment (jamalex) [inline kolibri/core/auth/management/commands/deletefacility.py:203]: *now?

**Review by jamalex — COMMENTED** [review]
Sorry for not reviewing before -- looks great! Just added a couple comment typo notes and a random comment. Nothing needing action.

**Pull request merged**